docs/reference/types.md

```diff
@@ -0,0 +1,81 @@
+(types)=
+# Type format
+
+SQLair supports structs maps and slices in input expressions and structs and
+maps in output expressions. 
+
+All types used with SQLair must be named so they can be referenced in the query.
+
+## Structs
+
+To use a struct with SQLair, all struct fields that correspond to a database
+column must be tagged with a `db` tag containing the column name. These fields
+must be public (i.e. start with a capital letter), though the struct itself does
+not have to be. Any fields without a tag will be ignored.
+
+If a struct contains an embedded struct then SQLair will treat the fields of the
+embedded structs as if they were fields in the parent struct.
+
+For example:
+```go
+type Person struct {
+    Name    string  `db:"name"`
+    ID      int     `db:"id_number"`
+    DoB     string  `db:"date_of_birth"`
+    Weight  int
+    height  int
+}
+```
+In this example the `Weight` and `height` fields will be ignored by SQLair.
+
+### The "omitempty" keyword
+
+The `omitempty` keyword tells SQLair to omit a column specified in a struct tag
```

Comment:
I'd start with "In a struct tag, ..." -- a zoom-in approach generally makes things easier to parse (you start with something people already know, namely the struct and its tag, and then continue with the new info about this specific struct tag keyword).